docs: Add a commit message specification.

bwinton opened the pull request:
Cribbed from Angular, but with more full sentences.
Also, fix up the readme to point to it.

Reviewer: @ericawright

ericawright commented:
Notes:
- add link to ‘merge commits’ in top
- remove link to ’scope’ from top 
- are we going to be actively using `git bisect` in the script you'll write?
- it seems these instructions are written for all commits, it does not specify these are only for merging
  - since thers are only for merge commits/PRs do we need the ‘style’ type?
- under type, specify to not capitalize the <type>
- breaking changes in the footer, not the body -- do we want this? Also, do we need as descriptive as their example? Or do we need this at all since it auto-updates we don’t really have versioning issues?

bwinton commented:
> * add link to ‘merge commits’ in top

Fixed.

> * remove link to ’scope’ from top

Fixed.

> * are we going to be actively using git bisect in the script you'll write?

Probably not often, but for tracking down breakages, perhaps…

> * it seems these instructions are written for all commits, it does not specify these are only for merging

Hmm…  good point.  Fixed.

>   * since these are only for merge commits/PRs do we need the ‘style’ type?

I think yes, because we could make PRs that just clean stuff up, or PRs that only add docs (like this one!), and that would be good to note.

> * under type, specify to not capitalize the

Fixed.

> * breaking changes in the footer, not the body -- do we want this? Also, do we need as descriptive as their example? Or do we need this at all since it auto-updates we don’t really have versioning issues?

Well, sorta…  I’ve updated it, let me know what you think.

ericawright commented:
`Emojis are recommended, but not required.` <3
